src/Workspaces/Core/Portable/NamingStyles/NamingStyle.cs

```diff
@@ -314,6 +315,14 @@ private string FinishFixingName(string name)
                 words = name.Split(new[] { WordSeparator }, StringSplitOptions.RemoveEmptyEntries);
             }
 
+            if (CapitalizationScheme == Capitalization.AllLower && !string.IsNullOrEmpty(WordSeparator))
+            {
+                if (words.Count() == 1) // Only Split if words have not been split before 
```

Comment:
I think we can get away with making the `IEnumerable<string>` into `IList<string>` and then avoid the extension method. Yeah, it'll do the casting under the covers but no reason to waste the time.

Reply:
Actually, don't all of these return string arrays?

Reply:
Not sure I've left it as is for now